### DISCUSSION TITLE
Getting same response for different query param passed to request

### DISCUSSION THREAD

**Discussion by javauser05** [mockoon/mockoon#1480] [Help] [answered]
Below is GET route **api/suppliers**

1.
![image](https://github.com/user-attachments/assets/3bac7b91-b93a-421c-b7d3-f948f63e1b77)
response configured is **{  "messaage": "connectionClose=true&httpClient.SocketTimeout=30000&status=active" }**

2.
![image](https://github.com/user-attachments/assets/b9c0261f-ce02-4b33-bfd9-b454530f4c74)
response configured is **{  "message": "status=active" }**

When below request are hit, it should return response configured in respective response. However, it returns same response as **{ "message": "status=active" }**

**http://localhost:3004/api/suppliers?status=active**
**http://localhost:3004/api/suppliers?connectionClose=true&httpClient.SocketTimeout=30000&status=active**

Attached environment for reference.
[APIMock.json](https://github.com/user-attachments/files/16501083/APIMock.json)

Could you please help here, how to get response based on above request ?

**Comment by 255kb** [comment, marked as answer]
You need to escape the dot in the third rule: 
![image](https://github.com/user-attachments/assets/e91f854d-ef90-4fa3-b416-8de7cf71ff8e)

Use a dot when your query string is an object and you want to access a parameter: `?obj[prop]=value` --> rule: `obj.value`
If your query string contains a dot, you need to escape it in the rules. 
We are using [qs](https://www.npmjs.com/package/qs) to parse the query string but without the `allowDots` option.

[Relevant docs](https://mockoon.com/docs/latest/route-responses/dynamic-rules/#2-property-name-or-path)

**Reply by javauser05** [reply]
@255kb ,when I tried through 3rd party rest client (postman), it works.

However, as per your suggestion, when I tried hitting api through my application, am getting below exception in my application and there is no hit on mock server.
![image](https://github.com/user-attachments/assets/435c7bd6-d759-4a96-a8c6-d97ca10918a1)


Any idea what could be the reason? 

Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint:
**http://127.0.0.1:3004/api/suppliers?connectionClose=true&httpClient.SocketTimeout=30000&status=active**
due to: There are 1 parameters that couldn't be set on the endpoint. Check the uri if the parameters are spelt correctly and that they are properties of the endpoint. Unknown parameters=[{SocketTimeout=30000}]

**Reply by 255kb** [reply]
> when I tried through 3rd party rest client (postman), it works.

CORS issue can only happen in the browser. They won't happen in curl, Postman, etc. 

Regarding the Java error you got, it's a bit out of our scope. I guess one of the query parameters is not defined in your code (`SocketTimeout`). Probably the dot is an issue.

**Reply by javauser05** [reply]
@255kb ,Thanks for update.

**Reply by javauser05** [reply]
Seems to be env issue. Issue is resolved.
Thanks @255kb
